Author: Steve Kemp <skx@debian.org>
Description: mailsnarf does not parse mail correctly.
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=149330
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/mailsnarf.c
+++ b/mailsnarf.c
@@ -178,7 +178,7 @@
 	if (smtp->state != SMTP_DATA) {
 		while ((i = buf_index(&buf, "\r\n", 2)) >= 0) {
 			line = buf_tok(&buf, NULL, i + 2);
-			line->base[line->end] = '\0';
+			line->base[line->end-1] = '\0';
 			p = buf_ptr(line);
 			
 			if (strncasecmp(p, "RSET", 4) == 0) {
